components/prism-markup.js

```diff
@@ -108,23 +120,62 @@ Object.defineProperty(Prism.languages.markup.tag, 'addInlined', {
 			inside: Prism.languages[lang]
 		};
 
-		var def = {};
-		def[tagName] = {
-			pattern: RegExp(/(<__[\s\S]*?>)(?:<!\[CDATA\[(?:[^\]]|\](?!\]>))*\]\]>|(?!<!\[CDATA\[)[\s\S])*?(?=<\/__>)/.source.replace(/__/g, function () { return tagName; }), 'i'),
+		attributes = attributes || { attr: TAG_ATTR };
+
+		// /<<TAG><BEFORE>(\s(\s*<ATTR>)+)?\s*>/
+		var openingTag = '<'
+			+ tagName // <TAG>
+			+ '(?:' + (attributes.before || '') + ')' // <BEFORE>
+			+ '(?:\\s(?:\\s*(?:' + attributes.attr + '))+)?' // (\s(\s*<ATTR>)+)?
+			+ '\\s*>';
+		var closingTag = '</' + tagName + '>';
+
+		var pattern = '(' + openingTag + ')'
+			+ /(?:<!\[CDATA\[(?:[^\]]|\](?!\]>))*\]\]>|(?!<!\[CDATA\[)[\s\S])*?/.source
+			+ '(?=' + closingTag + ')';
+
+		var token = {
+			pattern: RegExp(pattern, 'i'),
 			lookbehind: true,
 			greedy: true,
 			inside: inside
 		};
 
-		Prism.languages.insertBefore('markup', 'cdata', def);
+		/** @type {Array} */
+		var existingTokens = Prism.languages.markup[tagName];
+		if (!existingTokens) {
+			// no existing tokens, so we can just insert the new one before 'cdata'
+			var def = {};
+			def[tagName] = [token];
+			Prism.languages.insertBefore('markup', 'cdata', def);
+		} else {
+			// there are some existing tokens for this tag.
+			var index = existingTokens.length;
+			if (before) {
+				// instead of appending this token, we might want to insert it before some existing ones
+				for (var i = 0, t; t = existingTokens[i++];) {
+					if (before.some(function (b) { return ('language-' + b) in t.inside; })) {
+						index = i;
+						break;
+					}
+				}
+			}
+			existingTokens.splice(index, 0, token);
+		}
 	}
-});
 
-Prism.languages.html = Prism.languages.markup;
-Prism.languages.mathml = Prism.languages.markup;
-Prism.languages.svg = Prism.languages.markup;
+	Object.defineProperties(Prism.languages.markup['tag'], {
+		addInlined: { value: addInlined },
+		TAG_ATTR: { value: TAG_ATTR },
+	});
+
+	Prism.languages.html = Prism.languages.markup;
+	Prism.languages.mathml = Prism.languages.markup;
+	Prism.languages.svg = Prism.languages.markup;
+
+	Prism.languages.xml = Prism.languages.extend('markup', {});
+	Prism.languages.ssml = Prism.languages.xml;
+	Prism.languages.atom = Prism.languages.xml;
+	Prism.languages.rss = Prism.languages.xml;
```

Comment:
I think we can flatten these two assignment series down to one line for each. See if that saves anything.

Reply:
30 bytes or about 15% down.